src/LinAlg/hiopLinAlgFactory.cpp

```diff
@@ -117,6 +130,14 @@ hiopVectorInt* LinearAlgebraFactory::create_vector_int(const std::string& mem_sp
   const std::string mem_space_upper = toupper(mem_space);
   if(mem_space_upper == "DEFAULT") {
     return new hiopVectorIntSeq(size);
+  } else if(mem_space_upper == "CUDA") {
+    #ifdef HIOP_USE_CUDA
+      return new hiopVectorIntCuda(size, mem_space_upper);
+    #else
+      assert(false && "requested memory space not available because Hiop was not"
+                      "built with CUDA support");
```

Comment:
Same comment about assert here, but also in other sections of code in this file

Reply:
created issue #577